### PR TITLE
support matrix system test in travis

### PR DESCRIPTION
**Pull request by UDLD**

### CODE REVIEW

**Comment by UDLD** [comment]
@yangzhenlong

File: scripts/run.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+if [ ${TEST_STYLE} -eq 1 ]; then
```
Comment (rainchen) [inline scripts/run.sh]: use TEST_TYPE=UT or TEST_TYPE=ST to make it easy understanding

File: scripts/run-system-test.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+DEBUG_LOG_PATH=./debug_log.txt
+# upgrade gradle to 5.0
+wget http://services.gradle.org/distributions/gradle-5.0-all.zip > /dev/null
+unzip gradle-5.0-all.zip > ${DEBUG_LOG_PATH}
+./gradle-5.0/bin/gradle -v
```
Comment (rainchen) [inline scripts/run-system-test.sh]: should install gradle in traivis.yml using before_script, then we can run system test in local machine

**Review by rainchen — APPROVED** [review]
please update in a new PR

**Comment by rainchen** [comment]
This branch is out-of-date with the base branch

File: scripts/run.sh [hunk 3]
```diff
@@ -1,11 +1,11 @@
 #!/usr/bin/env bash
 
-if [ ${TEST_STYLE} -eq 1 ]; then
+if [ ${TEST_STYLE} == UT ]; then
```
Comment (rainchen) [inline scripts/run.sh]: rename TEST_STYLE to TEST_TYPE

File: scripts/run.sh (hunk 1 above)
Comment (UDLD) [reply scripts/run.sh]: fixed

File: scripts/run-system-test.sh (hunk 2 above)
Comment (UDLD) [reply scripts/run-system-test.sh]: fixed

File: scripts/run.sh (hunk 3 above)
Comment (UDLD) [reply scripts/run.sh]: fixed

**Comment by rainchen** [comment]
@UDLD This branch is out-of-date with the base branch

File: .travis.yml [hunk 4]
```diff
@@ -5,20 +5,33 @@ language: java
...
 # Avoid uploading cache after every build with Gradle - see https://docs.travis-ci.com/user/languages/java/#Caching
 before_cache:
   - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
   - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
 cache:
   directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
+  - $HOME/.gradle/caches/
```
Comment (duanyytop) [inline .travis.yml:24]: Why delete tab ?

**Review by duanyytop — APPROVED** [review]

**Pull request merged**